### PR TITLE
Fix #1216, Update Directory Diagrams in App Developer Guide

### LINKED ISSUE #1216: Update directory diagrams in application developer guide
**Is your feature request related to a problem? Please describe.**
After PR #1203 the directory structure diagrams in the user guide are in need of an update.  Only the text was updated in that PR.

**Describe the solution you'd like**
Diagrams should be updated to match the text.

**Additional context**
See #1203

**Requester Info**
Joseph Hickey, Vantage Systems, Inc.

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Describe the contribution**
Fixes #1216 

**Expected behavior changes**
Used tree to provide a tree view of cFS, cFE, Build, and Apps. 

**Additional context**
Did not add CS directory. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

File: docs/cFE Application Developers Guide.md [hunk 1]
```diff
@@ -303,16 +303,138 @@ a variation on this standard.
...
+    |-- build
+    |   |-- The flight software is all configured and built under this directory. 
+    |   |-- All mission and platform configuration files are placed here. 
+    |-- apps
+    |   |-- Contains application source code. 
+    |   |-- Application source code may be shared amoung multiple build CPUs. 
+
+-- missionxyz/cfe
+   |-- cmake
+   |   |-- sample_defs
```
[the remark this answers is not in the document]
Comment (ArielSAdamsNASA) [reply docs/cFE Application Developers Guide.md:333]: Added description for sample_defs.

File: docs/cFE Application Developers Guide.md [hunk 2]
```diff
@@ -303,16 +303,138 @@ a variation on this standard.
...
+   |       |-- sb
+   |       |-- sbr
+   |       |-- tbl
+   |       |-- time
+   |-- cpuN
+   |-- docs
+   |-- exe
+   |-- inc
+   |-- src
+   |-- tools
```
[the remark this answers is not in the document]
Comment (ArielSAdamsNASA) [reply docs/cFE Application Developers Guide.md]: Created a new tree for cpu1.

File: docs/cFE Application Developers Guide.md [hunk 3]
```diff
@@ -303,16 +303,138 @@ a variation on this standard.
...
+    |-- apps
+    |   |-- Contains application source code. 
+    |   |-- Application source code may be shared amoung multiple build CPUs. 
+
+-- missionxyz/cfe
+   |-- cmake
+   |   |-- sample_defs
+   |   |-- target
+   |       |-- inc
+   |       |-- src
```
[the remark this answers is not in the document]
Comment (ArielSAdamsNASA) [reply docs/cFE Application Developers Guide.md]: Replaced.

**Review by jphickey — APPROVED** [review]
Good improvement!

As an aside, I've found the `pstree` command line tool to make some nice cut-and-paste-able output for this type of thing.

**Comment by astrogeco** [comment]
[CCB:2021-09-01](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-September-01-2021) APPROVED

**Pull request merged**